src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -414,6 +416,18 @@ def set_raw_command_name(command):
     _session.raw_command = command
 
 
+@decorators.suppress_all_exceptions()
+def add_dedicated_instrumentation_key(dedicated_instrumentation_key):
+    if not dedicated_instrumentation_key:
+        return
+
+    from collections import Iterable
```

Comment:
I was using python 3.10 and this was silently erroring out because collections.Iterable was deprecated. I had to do something like this: 

```
    try:
        from collections.abc import Iterable
    except ImportError:
        from collections import Iterable
```